includes/class-newspack-newsletters.php

```diff
@@ -277,18 +277,15 @@ public static function rest_api_init() {
 		);
 		\register_rest_route(
 			'newspack-newsletters/v1',
-			'mailchimp/(?P<id>[\a-z]+)/interest/(?P<interest_id>[\a-z]+)',
+			'mailchimp/(?P<id>[\a-z]+)/segments',
 			[
 				'methods'             => \WP_REST_Server::EDITABLE,
-				'callback'            => [ __CLASS__, 'api_set_mailchimp_interest' ],
+				'callback'            => [ __CLASS__, 'api_set_mailchimp_segments' ],
 				'permission_callback' => [ __CLASS__, 'api_authoring_permissions_check' ],
 				'args'                => [
-					'id'          => [
+					'id' => [
 						'sanitize_callback' => 'absint',
 					],
-					'interest_id' => [
```

Comment:
Why is this arg removed? It's still being used in https://github.com/Automattic/newspack-newsletters/pull/241/commits/e15b9e734c8a1409a52dfc66ef1282bab89bdbc4#diff-76aefac9a481217d439a23e638846effR674

Reply:
Fixed in 90d99c86765f6afd243ede7f5c4d9be397aec354, for some reason I thought that if the payload is in the request body (as opposed to URL param), listing args is not needed.